sources/packages/backend/apps/workers/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -333,6 +334,107 @@ export class StudentAssessmentService extends RecordDataModelService<StudentAsse
     return true;
   }
 
+  /**
+   * Updates the last reported assessment for the current assessment.
+   * @param assessmentId related assessment id.
+   * @param entityManager entity manager to be part of the transaction.
+   * @returns updated assessment.
+   */
+  async updateLastReportedAssessment(
+    assessmentId: number,
+    entityManager: EntityManager,
+  ): Promise<UpdateResult | undefined> {
+    const assessment = await this.repo.findOne({
```

Comment:
please use the entity manager repo. Using global repo inside entity manager had issues as per our observation from Workers load test.